test/cypress/integration/widgets/filters_spec.ts

```diff
@@ -63,184 +101,612 @@ describe('Filters widget', () => {
+            cy.deleteDeployments(testPrefix)
+                .deleteBlueprints(testPrefix)
+                .uploadBlueprint('blueprints/empty.zip', blueprintId)
+                .deployBlueprint(blueprintId, deploymentId)
+                .setLabels(deploymentId, [
+                    { os: 'linux' },
+                    { os: 'windows' },
+                    { infra: 'aws' },
+                    { infra: 'gcp' },
+                    { infra: 'azure' }
+                ])
+                .deleteSites(testPrefix)
+                .createSites([
+                    { name: `${testPrefix}_Cracow` },
+                    { name: `${testPrefix}_Warsaw` },
+                    { name: `${testPrefix}_TelAviv` }
+                ]);
+        });
 
-        cy.contains('There are no filters defined');
-    });
+        beforeEach(openAddFilterModal);
+        afterEach(closeFilterModal);
 
-    it('should prevent filter used as default from being removed', () => {
-        cy.intercept('GET', `/console/filters/usage/${filterName}`, [
-            { pageName: 'Dashboard', widgetName: 'Deployments View', username: 'admin' }
-        ]).as('usageRequest');
+        function selectRuleRowType(ruleRowType: FilterRuleRowType) {
+            withinLastRuleRow(() => {
+                cy.get('div[name="ruleRowType"]').click();
+                cy.get(`div[option-value="${ruleRowType}"]`).click();
+            });
+        }
 
-        cy.get('.trash').click();
-        cy.contains('Yes').click();
+        function selectRuleOperator(operator: FilterRuleOperator) {
+            withinLastRuleRow(() => {
+                cy.get('div[name="ruleOperator"]').click();
+                cy.get(`div[option-value="${operator}"]`).click();
+            });
+        }
+
+        type RuleValueObject = { value: string; isNew: boolean };
+        function selectRuleAttributeValues(
+            ruleRowType: FilterRuleRowType,
+            values: RuleValueObject[],
+            withAutocomplete = false
+        ) {
+            if (values.length === 0) return;
+            withinLastRuleRow(() => {
+                const searchEndpoint: Record<FilterRuleRowType, string> = {
+                    blueprint_id: 'blueprints',
+                    site_name: 'sites',
+                    created_by: 'users',
+                    label: '' // NOTE: Only endpoints for attribute rules are necessary
+                };
+                cy.get('div[name="ruleValue"]')
+                    .click()
+                    .within(() => {
+                        values.forEach(({ value, isNew }) => {
+                            if (withAutocomplete) {
+                                const searchEndpointRegExp = RegExp(`${searchEndpoint[ruleRowType]}.*_search=${value}`);
+                                cy.interceptSp('GET', searchEndpointRegExp).as(`valueSearch_${value}`);
+                            }
+
+                            cy.get('input').type(`${value}`);
```

Comment:
Ha! I've found something to improve in the next PR

```suggestion
                            cy.get('input').type(value);
```

Reply:
Oh, I thought I already change that. Will fix in the next PR definitely 👍🏻